CalendarApp/Notification Handler/NotificationHandler.swift

```diff
@@ -0,0 +1,160 @@
+                guard let reminderID = notifications[0].reminderID else {
+                    return
+                }
+                center.removePendingNotificationRequests(withIdentifiers: [reminderID.uuidString])
+                scheduleNotification(event: event, date: date)
+                try context.save()
+            }
+        } catch {
+            // TODO: error handling
+        }
+    }
+    
+    func deleteReminderForEvent(_ eventID: UUID) {
+        let request = Reminder.fetchRequest()
+        request.predicate = NSPredicate(format: "eventID == %@",
+                                        eventID as CVarArg)
+        do {
+            let notifications = try context.fetch(request)
+            if notifications.count == 1 {
+                context.delete(notifications[0])
+                guard let reminderID = notifications[0].reminderID else {
+                    return
+                }
+                center.removePendingNotificationRequests(withIdentifiers: [reminderID.uuidString])
+                try context.save()
+            }
+        } catch {
+            // TODO: error handling
+        }
+    }
+    
+    func checkReminderForEvent(_ eventID: UUID) -> Date? {
+        let request = Reminder.fetchRequest()
+        request.predicate = NSPredicate(format: "eventID == %@",
+                                        eventID as CVarArg)
+        do {
+            let notifications = try context.fetch(request)
+            if notifications.count == 1 {
+                return notifications[0].reminderDate;
+            }
+        } catch {
+            // TODO: error handling
+        }
+        return nil
+    }
+    
+    func scheduleNotification(event: Event, date: Date) {
+        let content = UNMutableNotificationContent()
+        content.title = event.eventTitle + " Event"
+        content.body = getDateString(event: event, date: date)
+        content.categoryIdentifier = "alarm"
+        content.sound = UNNotificationSound.default
+        
+        let dateComponents = calendar.dateComponents([.year, .month, .day, .hour, .minute], from: date)
+        let trigger = UNCalendarNotificationTrigger(dateMatching: dateComponents, repeats: false)
+        
+        let notificationID = UUID()
+        let request = UNNotificationRequest(identifier: notificationID.uuidString, content: content, trigger: trigger)
+        
+        center.add(request)
+        
+        let eventReminder = Reminder(context: context)
+        eventReminder.eventID = event.objectUUID
+        eventReminder.reminderID = notificationID
+        eventReminder.reminderDate = date
+        eventReminder.archived = false
+        eventReminder.reminderDescription = content.body
+        eventReminder.title = content.title
+        do {
+            try context.save()
+        } catch {
+            // TODO: error handling
+        }
+    }
+    
+    private func getDateString(event: Event, date: Date) -> String {
```

Comment:
Something like this.. I get why you put String in the name but really it's more about the formatting. In this case I would probably try to name it that way. `formattedDate(event: Event, date: Date) -> String`. Or maybe `makeNotificationBody(event: Event, date: Date) -> String` ?